### PR TITLE
move assets to sightings

### PR DESCRIPTION
**Pull request by colinwkingen**
## Pull Request Overview

Attempt to move the asset list from encounter to sighting.

--

### **Review Notes**

--

This is mostly moved code, including tests. A new single migration is generated for the whole database build.

Review should mostly be making sure I didn't leave any misleading asset related comments or code around the encounter
module. 

### **API Changes**

This moves PATCH calls with op **assetId** or **newAssetGroup** from /api/v1/encounters/ to /api/v1/sightings/.  

**Database Updates [OPTIONAL]**

Moving the asset list field from encounter to sighting. 

**Invoke CLI Updates [OPTIONAL]**

**Pull Request Checklist**
- [x] Ensure that the PR is properly formatted
  - Example: All lint checks are passing
- [x] Ensure that the PR is properly rebased
  - Example: The PR is rebased on `develop` (commit: `<insert develop commit hash>`)
- [x] Ensure that the PR uses a consolidated database migration
  - Example: One database migration is proposed (revision `<insert develop revision>` -> `<insert new revision>`)
- [x] Ensure that the PR is properly tested
  - Example: All automated tests are passing
- [x] Ensure that the PR is properly covered
  - Example: The percentage of the code covered by tests has not decreased
- [x] Ensure that the PR is properly sanitized
  - Example: No sensitive data or large content was added to this PR
- [ ] Ensure that the PR is properly reviewed

### CODE REVIEW

**Comment by colinwkingen** [comment]
Hey @hwindsor @naknomum  would appreciate an eye on why my last test is failing. To patch an asset on to an existing sighting **test_asset_file_addition** fails with the response:

`{"status": 422, "messages": {"0": {"op": ["Not a valid choice."], "path": ["Not a valid choice."], "_schema": ["operation not supported"]}}, "message": "The request was well-formed but was unable to be followed due to semantic errors."}
`

I can't fathom why, the path is added to the available choices in parameters.py, OP_ADD is allowed and the code to do it is largely just moved over from the old encounter section. I've tried a bunch of tweaks locally but the code here is basically what I've reverted to. I'm hoping it's something silly that will jump out to one of you

File: app/modules/sightings/models.py [hunk 1]
```diff
@@ -21,6 +28,8 @@ class Sighting(db.Model, FeatherModel):
     )  # pylint: disable=invalid-name
     version = db.Column(db.BigInteger, default=None, nullable=True)
 
+    assets = db.relationship('SightingAssets')
```
Comment (hwindsor) [inline app/modules/sightings/models.py:31]: Is this necessary? Surely this is created by line 17 above with the back_populates?

File: app/modules/sightings/resources.py [hunk 2]
```diff
@@ -376,13 +366,14 @@ def post(self, args):
...
                 'encounters': result_data['encounters'],
+                'assets': assets,
             },
         }
         return rtn
 
 
 @api.route('/<uuid:sighting_guid>')
-@api.login_required(oauth_scopes=['sightings:read'])
+@api.login_required(oauth_scopes=[])
```
Comment (hwindsor) [inline app/modules/sightings/resources.py:376]: huh?

File: tests/modules/sightings/resources/test_create_sighting.py [hunk 3]
```diff
@@ -21,7 +21,7 @@ def test_create_failures(flask_app_client, test_root, researcher_1):
     assert not response.json['success']
 
     # has encounters, zero assetReferences, but fails on bad (missing) context value
-    data_in = {'encounters': [{'taxonomy': {'id': '0000000'}}]}
+    data_in = {'encounters': [{}]}
```
Comment (hwindsor) [inline tests/modules/sightings/resources/test_create_sighting.py:24]: why not? Surely this interface hasn't changed... yet?

File: tests/modules/sightings/resources/test_sighting_assets.py [hunk 4]
```diff
@@ -0,0 +1,155 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=missing-docstring
+from tests import utils
+
+from flask import current_app
+
+from tests.modules.sightings.resources import utils as sighting_utils
+
+
+PATH = '/api/v1/sightings/'
```
Comment (hwindsor) [inline tests/modules/sightings/resources/test_sighting_assets.py:10]: Don't think this is used.

File: app/modules/sightings/models.py (hunk 1 above)
Comment (colinwkingen) [reply app/modules/sightings/models.py:31]: Well, I wouldn't have though so but sqlalchemy seems to think it is, even with a fresh migration:
`sqlalchemy.exc.InvalidRequestError: Mapper 'mapped class Sighting->sighting' has no property 'assets'`

File: tests/modules/sightings/resources/test_create_sighting.py (hunk 3 above)
Comment (colinwkingen) [reply tests/modules/sightings/resources/test_create_sighting.py:24]: No, but that bit isn't used yet. If you don't have a matching taxonomy (and there isn't a taxonomy '0000000' in my local) then the EDM gets mad. Didn't seem relevant to what I was testing to add default taxonomies and validate them.

When we start getting more into validating encounter and sighting info I think we'll need to push more test data like this into the EDM database as a first step. Actually sort of looking at that for IA config lists elsewhere.

File: app/modules/sightings/resources.py (hunk 2 above)
Comment (colinwkingen) [reply app/modules/sightings/resources.py:376]: It's at the top of SightingByID, the thought was to ensure that anything below required a login, but to leave the scope to the individual ops if they choose to define it. If you really don't like it I can move it down, ha!

Comment (hwindsor) [reply app/modules/sightings/resources.py:376]: It just seemed odd. But the upshot is now the get doesn't require 'sightings:read' which I think is a bug.
Not an expert but it seems strange to me to have the login required and the auth check at this level for any resources class as the permission is different for get/patch/post/delete. 
I think these should all be removed from the class level and moved to the method level.

**Comment by colinwkingen** [comment]
@hwindsor Removing the **login_required** check at the beginning of **SightingByID** brought to light a small bug that we might need to look for elsewhere. 

The ops inside each had permission_required defined before login required. We had a test that checked for a 401 if a not-logged-in attempt was made to patch a sighting, but since it hit _permission_ required first it received a 403. Might pop up again elsewhere if we remove the top level checks in favor of more granular ones.

**Pull request merged**